lib/shopify_api/graphql.ex

```diff
@@ -0,0 +1,119 @@
+defmodule ShopifyAPI.GraphQL do
+  @moduledoc """
+  Internal interface to Shopify's GraphQL Admin API, built on HTTPoison. 
+
+  Makes requests against Shopify GraphQL and returns a tuple containig
+  a %Response struct with %{response, metadata, status_code}
+
+  * `query/2`
+
+  iex> query = "mutation {
+    metafieldDelete (
+            input: {
+              id: \"gid://shopify/Metafield/1234567891234\"
+            }
+    )
+    {
+      deletedId
+      userErrors {
+        field
+        message
+      }
+      }
+  }"
+  iex> ShopifyAPI.GraphQL.query(auth, query)
+  {:ok, %Response{
+          response: %{
+            "metafield1" => %{
+              "deletedId" => "gid://shopify/Metafield/5256098316335"
+            }
+          },
+          metadata: %{
+            "cost" => %{
+              "actualQueryCost" => 10,
+              "requestedQueryCost" => 10,
+              "throttleStatus" => %{
+                "currentlyAvailable" => 990
+              }
+            }
+          }, 
+          status_code: 200
+        }
+  }
+
+  """
+
+  require Logger
+
+  alias ShopifyAPI.AuthToken
+  alias ShopifyAPI.GraphQL.{Response, Telemetry}
+
+  # Use HTTP in test for Bypass, HTTPS in all other environments
+  @transport if Mix.env() == :test, do: "http://", else: "https://"
+
+  @default_graphql_version "2019-07"
+
+  @log_module __MODULE__ |> to_string() |> String.trim_leading("Elixir.")
+
+  def query(%AuthToken{} = auth, query, opts \\ []) do
+    url = build_url(auth, opts)
+    headers = build_headers(auth, opts)
+
+    logged_request(auth, url, query, headers, opts)
+  end
+
+  def configured_version do
+    config = Application.get_env(:shopify_api, ShopifyAPI.GraphQL, [])
+    Keyword.get(config, :graphql_version, @default_graphql_version)
+  end
+
+  def remaining_points({:ok, %Response{metadata: metadata}}) do
+    metadata["cost"]["throttleStatus"]["currentlyAvailable"]
```

Comment:
This might be a good place to use `get_in/2`, which handles the case where `metadata == %{}`:

```elixir
get_in(metadata, ["cost", "throttleStatus", "currentlyAvailable"])
```